src/debug/debug_impl.ts

```diff
@@ -860,7 +860,7 @@ class ThreadManager {
 
 	removeStoredIds(ids: number[]) {
 		for (let id of ids) {
-			delete this.storeData[id];
```

Comment:
That does look weird, I suspect you're right. I'm not totally familiar with everything in this file as @devoncarew contributed it and it's rather complicated to debug (the irony!). I think Code has incoming changes that might make that easier (debugger attached to multiple processes - the extension and the debugger).